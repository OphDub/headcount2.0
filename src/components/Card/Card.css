article {
    border: 1px solid lightslategray;
    height: 325px;
    width: 200px;
    margin: 10px;
    font-family: 'Francois One', sans-serif;
}

ul {
    margin: auto;
    padding-left: 60px;
    list-style-type: none;
}

h4 {
    margin: auto;
    margin-bottom: 20px;
    padding: 10px;
    text-align: center;
    background-color: rgb(74, 169, 241);
    color: rgb(232, 238, 241);
}

.highFive {
    color: blue;
}

.lowFive {
    color: orangered;
}

.cardContainer article:hover {
    box-shadow: 4px 4px 0px 0px rgba(22, 22, 87, 0.2);
}

.compareContainer.selected {
  border: 3px solid rgb(57, 185, 231);
}

.selected {
    border: 3px solid rgb(57, 185, 231);
}