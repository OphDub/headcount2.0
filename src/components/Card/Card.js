import React from 'react';
import { string, object } from 'prop-types';
import './Card.css';

const Card = ({ location, data }) => {
  // <article onClick={() => selectCard(location)}> ?
  const eachCard = Object.keys(data).map( dataPoint => {
                                            const highOrLow = data[ dataPoint ] >= 0.5 ? "highFive" : "lowFive";
                                            
                                            return <li className={ highOrLow } >{ dataPoint + ': ' + data[ dataPoint ] }</li>
                                        } )
  return (
    <article>
      <h4>{ location }</h4>
      <ul>{ eachCard }</ul>
    </article>
  )
}

Card.propTypes = {
  location: string.isRequired,
  data: object.isRequired,
}

export default Card;
