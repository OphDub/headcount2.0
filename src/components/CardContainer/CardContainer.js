import React from 'react';
import { arrayOf, shape, string, object } from 'prop-types';
import './CardContainer.css';
import Card from '../Card/Card';

const CardContainer = ({ districts, selectCard, comparedDistricts }) => {
  const addCss = (district) => {
    const matched = comparedDistricts.find( card => {
      return card.location === district.location
    })

    return matched ? "selected" : "";
  }

  const renderedCards = districts.map((district) =>
        <Card {...district}
               key={district.location}
               id={district.location}
               selectCard={selectCard}
               addCss={addCss(district)}/>)

  return (
    <section className="cardContainer">
      {renderedCards}
    </section>
  )
}

CardContainer.propTypes = {
  districts: arrayOf(shape({
    location: string.isRequired,
    data: object.isRequired,
  }))
}

export default CardContainer;
