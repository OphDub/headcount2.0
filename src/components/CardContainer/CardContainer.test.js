import React from 'react';
import ReactDOM from 'react-dom';
import CardContainer from './CardContainer';
import Card from '../Card/Card';
import { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

describe('CARD CONTAINER', () => {

  test.skip('it should match the snapshot', () => {
    const renderedComponent = shallow(<CardContainer />);
    expect(renderedComponent).toMatchSnapshot();
  })
})