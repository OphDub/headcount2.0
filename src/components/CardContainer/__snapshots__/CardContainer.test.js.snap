// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CARD CONTAINER should match snap of cardContainer 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    addCss={[Function]}
    districts={
        Array [
            Object {
              "data": Object {
                "2004": 0,
                "2005": 0,
                "2006": 1,
                "2007": 1,
                "2008": 1,
                "2009": 1,
                "2010": 1,
                "2011": 1,
                "2012": 1,
                "2013": 1,
                "2014": 1,
              },
              "location": "CAMPO RE-6",
            },
            Object {
              "data": Object {
                "2004": 0.302,
                "2005": 0.353,
                "2006": 0.119,
                "2007": 0.218,
                "2008": 0.766,
                "2009": 0.98,
                "2010": 0.993,
                "2011": 0.993,
                "2012": 1,
                "2013": 1,
                "2014": 1,
              },
              "location": "CANON CITY RE-1",
            },
            Object {
              "data": Object {
                "2004": 0,
                "2005": 0.105,
                "2006": 0.045,
                "2007": 0,
                "2008": 1,
                "2009": 1,
                "2010": 1,
                "2011": 1,
                "2012": 1,
                "2013": 1,
                "2014": 1,
              },
              "location": "CENTENNIAL R-1",
            },
            Object {
              "data": Object {
                "2004": 1,
                "2005": 1,
                "2006": 1,
                "2007": 1,
                "2008": 1,
                "2009": 1,
                "2010": 1,
                "2011": 1,
                "2012": 1,
                "2013": 1,
                "2014": 1,
              },
              "location": "CENTER 26 JT",
            },
          ]
    }
    selectCard={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          addCss={[Function]}
          data={
                    Object {
                              "2004": 0,
                              "2005": 0,
                              "2006": 1,
                              "2007": 1,
                              "2008": 1,
                              "2009": 1,
                              "2010": 1,
                              "2011": 1,
                              "2012": 1,
                              "2013": 1,
                              "2014": 1,
                            }
          }
          id="CAMPO RE-6"
          location="CAMPO RE-6"
          selectCard={[Function]}
/>,
        <Card
          addCss={[Function]}
          data={
                    Object {
                              "2004": 0.302,
                              "2005": 0.353,
                              "2006": 0.119,
                              "2007": 0.218,
                              "2008": 0.766,
                              "2009": 0.98,
                              "2010": 0.993,
                              "2011": 0.993,
                              "2012": 1,
                              "2013": 1,
                              "2014": 1,
                            }
          }
          id="CANON CITY RE-1"
          location="CANON CITY RE-1"
          selectCard={[Function]}
/>,
        <Card
          addCss={[Function]}
          data={
                    Object {
                              "2004": 0,
                              "2005": 0.105,
                              "2006": 0.045,
                              "2007": 0,
                              "2008": 1,
                              "2009": 1,
                              "2010": 1,
                              "2011": 1,
                              "2012": 1,
                              "2013": 1,
                              "2014": 1,
                            }
          }
          id="CENTENNIAL R-1"
          location="CENTENNIAL R-1"
          selectCard={[Function]}
/>,
        <Card
          addCss={[Function]}
          data={
                    Object {
                              "2004": 1,
                              "2005": 1,
                              "2006": 1,
                              "2007": 1,
                              "2008": 1,
                              "2009": 1,
                              "2010": 1,
                              "2011": 1,
                              "2012": 1,
                              "2013": 1,
                              "2014": 1,
                            }
          }
          id="CENTER 26 JT"
          location="CENTER 26 JT"
          selectCard={[Function]}
/>,
      ],
      "className": "cardContainer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "CAMPO RE-6",
        "nodeType": "function",
        "props": Object {
          "addCss": [Function],
          "data": Object {
            "2004": 0,
            "2005": 0,
            "2006": 1,
            "2007": 1,
            "2008": 1,
            "2009": 1,
            "2010": 1,
            "2011": 1,
            "2012": 1,
            "2013": 1,
            "2014": 1,
          },
          "id": "CAMPO RE-6",
          "location": "CAMPO RE-6",
          "selectCard": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "CANON CITY RE-1",
        "nodeType": "function",
        "props": Object {
          "addCss": [Function],
          "data": Object {
            "2004": 0.302,
            "2005": 0.353,
            "2006": 0.119,
            "2007": 0.218,
            "2008": 0.766,
            "2009": 0.98,
            "2010": 0.993,
            "2011": 0.993,
            "2012": 1,
            "2013": 1,
            "2014": 1,
          },
          "id": "CANON CITY RE-1",
          "location": "CANON CITY RE-1",
          "selectCard": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "CENTENNIAL R-1",
        "nodeType": "function",
        "props": Object {
          "addCss": [Function],
          "data": Object {
            "2004": 0,
            "2005": 0.105,
            "2006": 0.045,
            "2007": 0,
            "2008": 1,
            "2009": 1,
            "2010": 1,
            "2011": 1,
            "2012": 1,
            "2013": 1,
            "2014": 1,
          },
          "id": "CENTENNIAL R-1",
          "location": "CENTENNIAL R-1",
          "selectCard": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "CENTER 26 JT",
        "nodeType": "function",
        "props": Object {
          "addCss": [Function],
          "data": Object {
            "2004": 1,
            "2005": 1,
            "2006": 1,
            "2007": 1,
            "2008": 1,
            "2009": 1,
            "2010": 1,
            "2011": 1,
            "2012": 1,
            "2013": 1,
            "2014": 1,
          },
          "id": "CENTER 26 JT",
          "location": "CENTER 26 JT",
          "selectCard": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            addCss={[Function]}
            data={
                        Object {
                                    "2004": 0,
                                    "2005": 0,
                                    "2006": 1,
                                    "2007": 1,
                                    "2008": 1,
                                    "2009": 1,
                                    "2010": 1,
                                    "2011": 1,
                                    "2012": 1,
                                    "2013": 1,
                                    "2014": 1,
                                  }
            }
            id="CAMPO RE-6"
            location="CAMPO RE-6"
            selectCard={[Function]}
/>,
          <Card
            addCss={[Function]}
            data={
                        Object {
                                    "2004": 0.302,
                                    "2005": 0.353,
                                    "2006": 0.119,
                                    "2007": 0.218,
                                    "2008": 0.766,
                                    "2009": 0.98,
                                    "2010": 0.993,
                                    "2011": 0.993,
                                    "2012": 1,
                                    "2013": 1,
                                    "2014": 1,
                                  }
            }
            id="CANON CITY RE-1"
            location="CANON CITY RE-1"
            selectCard={[Function]}
/>,
          <Card
            addCss={[Function]}
            data={
                        Object {
                                    "2004": 0,
                                    "2005": 0.105,
                                    "2006": 0.045,
                                    "2007": 0,
                                    "2008": 1,
                                    "2009": 1,
                                    "2010": 1,
                                    "2011": 1,
                                    "2012": 1,
                                    "2013": 1,
                                    "2014": 1,
                                  }
            }
            id="CENTENNIAL R-1"
            location="CENTENNIAL R-1"
            selectCard={[Function]}
/>,
          <Card
            addCss={[Function]}
            data={
                        Object {
                                    "2004": 1,
                                    "2005": 1,
                                    "2006": 1,
                                    "2007": 1,
                                    "2008": 1,
                                    "2009": 1,
                                    "2010": 1,
                                    "2011": 1,
                                    "2012": 1,
                                    "2013": 1,
                                    "2014": 1,
                                  }
            }
            id="CENTER 26 JT"
            location="CENTER 26 JT"
            selectCard={[Function]}
/>,
        ],
        "className": "cardContainer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "CAMPO RE-6",
          "nodeType": "function",
          "props": Object {
            "addCss": [Function],
            "data": Object {
              "2004": 0,
              "2005": 0,
              "2006": 1,
              "2007": 1,
              "2008": 1,
              "2009": 1,
              "2010": 1,
              "2011": 1,
              "2012": 1,
              "2013": 1,
              "2014": 1,
            },
            "id": "CAMPO RE-6",
            "location": "CAMPO RE-6",
            "selectCard": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "CANON CITY RE-1",
          "nodeType": "function",
          "props": Object {
            "addCss": [Function],
            "data": Object {
              "2004": 0.302,
              "2005": 0.353,
              "2006": 0.119,
              "2007": 0.218,
              "2008": 0.766,
              "2009": 0.98,
              "2010": 0.993,
              "2011": 0.993,
              "2012": 1,
              "2013": 1,
              "2014": 1,
            },
            "id": "CANON CITY RE-1",
            "location": "CANON CITY RE-1",
            "selectCard": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "CENTENNIAL R-1",
          "nodeType": "function",
          "props": Object {
            "addCss": [Function],
            "data": Object {
              "2004": 0,
              "2005": 0.105,
              "2006": 0.045,
              "2007": 0,
              "2008": 1,
              "2009": 1,
              "2010": 1,
              "2011": 1,
              "2012": 1,
              "2013": 1,
              "2014": 1,
            },
            "id": "CENTENNIAL R-1",
            "location": "CENTENNIAL R-1",
            "selectCard": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "CENTER 26 JT",
          "nodeType": "function",
          "props": Object {
            "addCss": [Function],
            "data": Object {
              "2004": 1,
              "2005": 1,
              "2006": 1,
              "2007": 1,
              "2008": 1,
              "2009": 1,
              "2010": 1,
              "2011": 1,
              "2012": 1,
              "2013": 1,
              "2014": 1,
            },
            "id": "CENTER 26 JT",
            "location": "CENTER 26 JT",
            "selectCard": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
